zeppelin-server/src/test/java/org/apache/zeppelin/integration/SparkParagraphIT.java

```diff
@@ -0,0 +1,241 @@
+            "ERROR", CoreMatchers.equalTo("FINISHED")
+        );
+        LOG.error("Paragraph got ERROR");
+      }
+
+      WebElement paragraph2Result = driver.findElement(By.xpath(
+          getParagraphXPath(2) + "//div[@class=\"tableDisplay\"]"));
+
+      collector.checkThat("Paragraph result is ::",
+          paragraph2Result.getText().toString(), CoreMatchers.containsString(
+              "import org.apache.commons.io.IOUtils"
+          )
+      );
+
+    } catch (Exception e) {
+      handleException("Exception in SparkParagraphIT while testSpark", e);
+    }
+  }
+
+  @Test
+  public void testPySpark() throws Exception {
+    if (!endToEndTestEnabled()) {
+      return;
+    }
+    try {
+      WebElement paragraph1Editor = driver.findElement(By.xpath(getParagraphXPath(1) + "//textarea"));
+
+      paragraph1Editor.sendKeys(PERCENTAGE + "pyspark" + ENTER +
+          "for x in range" + OPEN_PARENTHESIS + "0, 3):" + ENTER +
+          "    print \"test loop " + PERCENTAGE + "d\" " +
+          PERCENTAGE + " " + OPEN_PARENTHESIS + "x)" + ENTER);
+
+      paragraph1Editor.sendKeys(SHIFT_ENTER);
+
+      try {
+        waitForParagraph(1, "FINISHED");
+      } catch (TimeoutException e) {
+        waitForParagraph(1, "ERROR");
+        WebElement paragraph1Result = driver.findElement(By.xpath(
+            getParagraphXPath(1) + "//div[@class=\"tableDisplay\"]"));
+        if (paragraph1Result.getText().toString().contains("pyspark 1.1.1 is not supported")) {
+          LOG.info("pyspark is not supported in spark version 1.1.x, nothing to worry.");
+        } else {
+          collector.checkThat("Paragraph result resulted in error ::",
+              "ERROR", CoreMatchers.equalTo("FINISHED")
+          );
+          LOG.error("Paragraph got ERROR");
+        }
+      }
+
+      WebElement paragraph1Result = driver.findElement(By.xpath(
+          getParagraphXPath(1) + "//div[@class=\"tableDisplay\"]"));
+
+      if (!paragraph1Result.getText().toString().contains("pyspark 1.1.1 is not supported")) {
+        collector.checkThat("Paragraph result is ::",
+            paragraph1Result.getText().toString(), CoreMatchers.equalTo("test loop 0\ntest loop 1\ntest loop 2")
+        );
+      }
+
+    } catch (Exception e) {
+      handleException("Exception in SparkParagraphIT while testPySpark", e);
+    }
+  }
+
+  @Test
+  public void testSqlSpark() throws Exception {
+    if (!endToEndTestEnabled()) {
+      return;
+    }
+    try {
+      WebElement paragraph1Editor = driver.findElement(By.xpath(getParagraphXPath(1) + "//textarea"));
+
+      paragraph1Editor.sendKeys(PERCENTAGE + "sql" + ENTER +
+          "select * from bank limit 1");
+
+      paragraph1Editor.sendKeys(SHIFT_ENTER);
+
+      try {
+        waitForParagraph(1, "FINISHED");
+      } catch (TimeoutException e) {
+        waitForParagraph(1, "ERROR");
+        collector.checkThat("Paragraph result resulted in error ::",
```

Comment:
would it be possible in the check and LOG.error to specify this is from the sql spark test?